src/main/java/net/mtrop/doom/tools/decohack/DecoHackParser.java

```diff
@@ -2658,6 +2692,31 @@ private Integer matchFlagMnemonic()
 			nextToken();
 		return out;
 	}
+	
+	// Matches an identifier that references a weapon
+	// flag mnemonic. If match, advance token and
+	// return bitflags. Else, return null.
+	private Integer matchWeaponFlagMnemonic()
+	{
+		if (!currentType(DecoHackKernel.TYPE_IDENTIFIER))
+			return null;
+		
+		Integer out = null;
+		String mnemonic = currentToken().getLexeme().toLowerCase();
+
+		for (int i = 0; i < DEHWeaponFlag.VALUES.length; i++)
+		{
+			if(DEHWeaponFlag.VALUES[i].getMnemonic().equals(mnemonic))
+			{
+				out = DEHWeaponFlag.VALUES[i].getValue();
+				break;
+			}
+		}
```

Comment:
There's definitely an improvement that can be done here by making a static name lookup map in `DEHWeaponFlag` but I'm fine with kicking that can down the road when I dig back into the parser, later.